--- a/mcs/tools/cil-stringreplacer/Makefile	2019-02-21 04:46:01.764018606 +0000
+++ b/mcs/tools/cil-stringreplacer/Makefile	2019-02-21 04:46:33.532559588 +0000
@@ -8,7 +8,7 @@
 API = $(filter basic build, $(PROFILE))
 ifdef API
 # It can be run using system .net during boostrap
-TARGET_NET_REFERENCE = v4.6
+TARGET_NET_REFERENCE = v4.7.1
 # Trick to make it work during boostrap where it has to run with system
 # assemblies not the ones in lib/basic folder
 PROGRAM_USE_INTERMEDIATE_FILE = 1
